Issue #50 robot model collision

jrgnicho opened the pull request:
This PR should be reviewed after #49 is accepted.
- It addresses the issue described in #50 regarding collision checks.
- In addition to that, it added two interface methods for the descartes_core::RobotModel  for setting and getting a map containing option key/value pairs.  
- The constructor for the MoveitStateAdapter no longer takes any initialization arguments, instead it takes them in the 'initialize(..)' interface method.
- The unit test have been updated to conform to the changes above.

jrgnicho committed:
added setter and getter for options map

jrgnicho committed:
checks for collisions inside the getIK(...) method

jrgnicho committed:
updated unit test for the MoveitStateAdapter class

shaun-edwards commented:
@jrgnicho, can you fix this PR.  The diff shows changes from already accepted PRs.

jrgnicho commented:
OK, will do

jrgnicho commented:
@shaun-edwards , the issue with the old diffs has been fixed and some minor changes for setting an options map into the MoveitStateAdapter have been made as well.  
Please review

jrgnicho committed:
added a 'check_collisions' option and tested it in the unit test

jrgnicho commented:
@shaun-edwards or @Jmeyer1292  ping

jrgnicho committed:
removed the setter and getter methods for the options map and added setters and getters for checking collisions

jrgnicho commented:
@shaun-edwards I added setter and getter methods to the RobotModel interface for enabling and disabling collision checks and removed the options map.  Please review

shaun-edwards commented on descartes_moveit/include/descartes_moveit/moveit_state_adapter.h:
```diff
@@ -85,7 +86,16 @@ class MoveitStateAdapter : public descartes_core::RobotModel
    */
+
+  /**
+   * TODO: Checks for collisions at this joint pose. The setCollisionCheck(true) must have been
```
This seems counter-intuitive.  I can see using the flag to turn on/off collision checking for IK, but anybody calling this method should expect it to perform collision checking.
<EDIT>
Should we consider adding this method to the base class?

jrgnicho replied:
This is a protected method so it is not meant to be called directly by the implementer.  I placed the collision code inside this method so that it can be conveniently invoked when needed by any of the interface methods.  There is already an **isValid** method which should also check for collisions when these are enabled; adding this method to the interface might make matters more confusing in my opinion.

shaun-edwards commented on descartes_core/include/descartes_core/robot_model.h:
```diff
@@ -102,9 +102,31 @@ class RobotModel
 protected:
 
-  RobotModel(){}
+  RobotModel(): check_collisions_(false){}
```
Should the default collision checking be `true`?  What do you think?

jrgnicho replied:
Since collision checks are expensive operations I think they should be disabled by default.

shaun-edwards commented:
Should we consider using the `isValid` base methods to check collisions as well.  This would require ensuring that their current use still makes sense.

shaun-edwards commented:
Couple of comments/items for discussion, but in general I think we should move forward.

jrgnicho commented:
The **isValid** method of the MoveitStateAdaptor in this PR checks for collisions when these are enabled.

jrgnicho commented:
@shaun-edwards  if there aren't any more issues to address here then this PR can be merged.

shaun-edwards commented:
+1
